R/join_key.R

```diff
@@ -2,33 +2,40 @@
 #'
 #' @description `r lifecycle::badge("stable")`
 #'
-#' @details `join_key()` will create a relationship for the variables on a pair
-#' of datasets.
+#' @description Create a relationship between two datasets, `dataset_1` and `dataset_2`.
+#' By default, this function establishes a directed relationship with `dataset_1` as the parent.
+#' If `dataset_2` is not specified, the function creates a primary key for `dataset_1`.
 #'
-#' @param dataset_1,dataset_2 (`character(1)`) dataset names. If `dataset_2` is omitted,
-#'  a primary key for `dataset_1` is created.
-#' @param keys (optionally named `character`) where `names(keys)` are columns in `dataset_1`
-#' corresponding to columns of `dataset_2` given by the elements of `keys`.
+#' @param dataset_1,dataset_2 (`character(1)`) Dataset names. When `dataset_2` is omitted,
+#' a primary key for `dataset_1` is created.
+#' @param keys (optionally named `character`) Column mapping between the datasets,
+#' where `names(keys)` maps columns in `dataset_1` corresponding to columns of
+#' `dataset_2` given by the elements of `keys`.
 #'
 #' If unnamed, the same column names are used for both datasets.
 #'
 #' If any element of the `keys` vector is empty with a non-empty name, then the name is
 #' used for both datasets.
+#' @param directed (`logical(1)`) Flag that indicates whether it should create
```

Comment:
I'm not sure `directed` is the best possible name.
Could we consider some alternatives?
`directional`
`hierarchical`
`parental`
`vertical`

Reply:
I think `directed` is a good name, but I'm not married. Maybe we can have a quick poll with emojis

React to this comment with:

:+1:  _for_ `directed`
:smile:  _for_ `directional`
:tada:  _for_ `hierarchical`
:heart:  _for_ `parental`
:rocket:  _for_ `vertical`
:eyes: _for_ `dataset_1_as_parent`

Reply:
"directed" is a legit term in a graph theory which join_keys follows. We also use other graph theory terms, like cyclic or topological sort. Directed is also short enough unlike "directional" or other proposed

Reply:
If it's a technical term (which I was unaware of), let's keep it, although I would question whether it is widely known.